data-prepper-core/src/test/java/org/opensearch/dataprepper/peerforwarder/RemotePeerForwarderTest.java

```diff
@@ -214,6 +219,37 @@ void test_receiveRecords_should_return_record_from_buffer() throws Exception {
         verify(recordsReceivedFromPeersCounter).increment(3.0);
     }
 
+    @Test
+    void test_receiveRecords_with_missing_identification_keys() throws Exception {
+        AggregatedHttpResponse aggregatedHttpResponse = mock(AggregatedHttpResponse.class);
+        when(aggregatedHttpResponse.status()).thenReturn(HttpStatus.OK);
+        when(peerForwarderClient.serializeRecordsAndSendHttpRequest(anyCollection(), anyString(), anyString(), anyString())).thenReturn(aggregatedHttpResponse);
+
+        final List<String> testIps = List.of("8.8.8.8", "127.0.0.1");
+        lenient().when(hashRing.getServerIp(List.of("value1", "value1"))).thenReturn(Optional.of(testIps.get(0)));
+        lenient().when(hashRing.getServerIp(List.of("value2", "value2"))).thenReturn(Optional.of(testIps.get(1)));
+
+        RemotePeerForwarder peerForwarder = createObjectUnderTest();
+        final Collection<Record<Event>> testRecords = generateBatchRecords(2);
+        // Add an event that doesn't have identification keys in it
+        final Map<String, String> eventData = new HashMap<>();
+        eventData.put("key3", "value3");
```

Comment:
Please generate random strings for these keys and values. They are arbitrary aside from having to not be equal to the existing identification keys. This will make the tests clearer.